src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -145,7 +146,20 @@ public string GetRelativeUrl(Document fileRelativeTo, Document file)
             // Cannot resolve the file, leave href as is
             if (file is null)
             {
-                return (error, href, fragment, hrefType, null);
+                if (string.IsNullOrEmpty(pathToDocset))
```

Comment:
how about putting this check into `TryResolveResourceFromHistory`, so that it can return null `resourceFromHistory`

Reply:
I am trying to keep it consistent with TryResolveContentFromHistory

Reply:
do not quite get the consistency here